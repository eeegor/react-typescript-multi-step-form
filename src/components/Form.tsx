/**
 * Component Form
 */

import * as React from 'react';
import * as classNames from 'classnames';
import './Form.scss';

export interface Props {
	id: string;
	className?: string;
	children?: React.ReactNode;
	onSubmit?: (form: React.FormEvent<HTMLFormElement>) => void;
}

export interface State {}

export class Form extends React.Component<Props, State> {
	constructor(props: Props) {
		super(props);
	}

	render(): JSX.Element {
		const { id, children, className, onSubmit } = this.props;
		const cssClass = 'form';
		const cssClasses = classNames(cssClass, className);
		const cssClassesContainer = classNames(`${cssClass}__container`);

		return (
			<form id={id} className={cssClasses} onSubmit={form => onSubmit && onSubmit(form)}>
				<div className={cssClassesContainer}>{children}</div>
			</form>
		);
	}
}
