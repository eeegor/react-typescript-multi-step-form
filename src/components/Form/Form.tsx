/**
 * Component Form
 */

import React from 'react';
import classnames from 'classnames';
import './Form.scss';

export interface Props {
	id: string;
	className?: string;
	children?: React.ReactNode;
	onSubmit?: (form: React.FormEvent<HTMLFormElement>) => void;
}

export const Form = (props: Props): JSX.Element => {
	const { id, children, className, onSubmit } = props;
	const cssClass = 'form';
	const cssClasses = classnames(cssClass, className);
	const cssClassesContainer = classnames(`${cssClass}__container`);

	return (
		<form
			id={id}
			className={cssClasses}
			onSubmit={form => onSubmit && onSubmit(form)}
			noValidate={true}
		>
			<div className={cssClassesContainer}>{children}</div>
		</form>
	);
};
