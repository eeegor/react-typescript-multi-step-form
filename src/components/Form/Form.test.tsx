import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Form } from './Form';

describe('Form', () => {
	it('renders markup correctly', () => {
		const wrapper = <Form id="first" />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('submit event works', () => {
		const callback = jest.fn();
		const wrapper = mount(
			<Form id="first" onSubmit={callback}>
				Hi
			</Form>
		);
		wrapper.find('form').simulate('submit');
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
