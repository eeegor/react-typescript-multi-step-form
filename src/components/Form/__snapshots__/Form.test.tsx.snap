// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders markup correctly 1`] = `
<form
  className="form"
  id="first"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="form__container"
  />
</form>
`;
